In [30]:
from sqlalchemy import create_engine, inspect
from sklearn_pandas import DataFrameMapper
import pandas as pd
from sklearn.preprocessing import LabelBinarizer
import psycopg2
import numpy as np

PSQL = 'postgres://{}@localhost:5432/rpred'

cred = ""
with open("../credentials/localhost/jessica.txt") as credfile:
    cred = credfile.read().strip("\n")
    
category_keywords = ['sandwiches', 'fast food', 'nightlife', 'pizza', 'bars',
                     'mexican', 'food', 'american', 'burgers', 'chinese',
                     'italian', 'breakfast & brunch', 'thai', 'indian',
                     'sushi', 'korean', 'mediterranean', 'japanese', 'seafood',
                     'middle eastern', 'pakistani', 'barbeque', 'vietnamese',
                     'asian fusion', 'diners', 'greek', 'vegetarian', 'cafes',
                    'comfort food', 'bakeries']
    
cnx = create_engine(PSQL.format(cred), isolation_level='AUTOCOMMIT')
df_c = pd.read_sql_query('''SELECT business_id, categories
FROM yelp_dataset_8''', cnx)

In [16]:
df_c.head()

business_id                                         categories
0  5UmKMjUEUNdYWqANhGckJw                       ['Fast Food'  'Restaurants']
1  yXuao0pFz1AxB21vJjDf5w                                ['Food'  'Grocery']
2  zaXDakTd3RXyOa7sMrUE1g                           ['Cafes'  'Restaurants']
3  8QlnAcjGE6dgfKTVoxrPvQ  ['American (Traditional)'  'Comfort Food'  'Ca...
4  PK6aSizckHFWk8i0oxt5DA            ['Burgers'  'Fast Food'  'Restaurants']

In [31]:
df_c['categories'] = df_c.categories.str.lower()
df_c.head()

business_id                                         categories
0  5UmKMjUEUNdYWqANhGckJw                       ['fast food'  'restaurants']
1  yXuao0pFz1AxB21vJjDf5w                                ['food'  'grocery']
2  zaXDakTd3RXyOa7sMrUE1g                           ['cafes'  'restaurants']
3  8QlnAcjGE6dgfKTVoxrPvQ  ['american (traditional)'  'comfort food'  'ca...
4  PK6aSizckHFWk8i0oxt5DA            ['burgers'  'fast food'  'restaurants']

# Find most popular keywords.

In [32]:
keyword_count = {}
for keyword in category_keywords:
    count = len(df_c[df_c.categories.str.contains(keyword)])
    keyword_count[keyword] = count


In [33]:
sorted(keyword_count.items(), key=lambda x: x[1], reverse=True)

[('food', 13486),
 ('fast food', 3154),
 ('sandwiches', 1220),
 ('bakeries', 1203),
 ('cafes', 1127),
 ('burgers', 963),
 ('bars', 901),
 ('mexican', 837),
 ('american', 808),
 ('breakfast & brunch', 622),
 ('nightlife', 408),
 ('chinese', 270),
 ('pizza', 206),
 ('italian', 148),
 ('seafood', 110),
 ('mediterranean', 103),
 ('comfort food', 101),
 ('barbeque', 72),
 ('vegetarian', 71),
 ('asian fusion', 61),
 ('thai', 59),
 ('diners', 58),
 ('indian', 55),
 ('greek', 55),
 ('japanese', 54),
 ('middle eastern', 54),
 ('vietnamese', 31),
 ('korean', 30),
 ('sushi', 28),
 ('pakistani', 8)]

## Top 10 interesting keywords:
1. ('fast food', 3154),
2. ('burgers', 963),
3. ('bars', 901),
4. ('mexican', 837),
5. ('american', 808),
6. ('chinese', 270),
7. ('pizza', 206),
8. ('italian', 148),
9. ('seafood', 110),
10. ('mediterranean', 103),